LoopKit/DeviceManager/CGMManager.swift

```diff
@@ -49,6 +57,12 @@ public protocol CGMManagerDelegate: DeviceManagerDelegate {
     /// - Parameter manager: The manager instance
     /// - Returns: The unique prefix for the credential store
     func credentialStoragePrefix(for manager: CGMManager) -> String
+    
+    /// Notifies the delegate of a change in status
+    ///
+    /// - Parameter manager: The manager instance
+    /// - Parameter status: The new, updated status
+    func cgmManager(_ manager: CGMManager, didUpdate status: CGMManagerStatus)
```

Comment:
I have to admit, this delegate is starting to get confusing.  There are three "did update" functions that seem very similar:

```swift
    func cgmManager(_ manager: CGMManager, didUpdateWith result: CGMResult) -> Void
```
```swift
    func cgmManagerDidUpdateState(_ manager: CGMManager)
```
...and now:
```swift
    func cgmManager(_ manager: CGMManager, didUpdate status: CGMManagerStatus)
```

Is there any way we could either coalesce them or rename them so they aren't so similar?

Reply:
Perhaps the most important part of reading that method signature is the noun. And each one is different, and should indicate the scope of the update. If you have suggestions to improve the clarity of those nouns, I'm open, but I think the nouns are there and are what should distinguish them.
 
* CGMResult = a reading from the sensor
* CGMManagerStatus = status of the manager; includes properties that are not associated with a sensor reading. Battery level might go here, for example, if we need to report that, or transmitter/sensor lifetime estimations.
* State = the opaque state that Loop is responsible for persisting, and using to restore the CGMManager with.

The verb (didUpdate) can stay the same, to indicate that this call is triggered by some kind of change, I think.

I'll update the inline documentation to describe some of these things as well.

Reply:
Ok.  You have to admit the nouns are also not very distinguishing ("Result", "Status", "State").  Maybe consider:
`CGMResult` -> `CGMSensorReading` 
`CGMManagerStatus` ->  `CGMStatus` (it really is the status of the CGM, not of the Manager, right?)
I'll leave it up to you to rename, if you want.  Adding clearer documentation to the `protocol` functions is a good idea, please do.

Reply:
Yeah, I strongly considered a `CGMResult` -> `CGMSensorReading`, but it's not a reading because it also handles errors and missing readings, and is used for responses to queries to get readings, so it's more like a result that might or might not contain a reading.  Maybe CGMSensorReadingResult? I think that fits what it is. It's a little wordy, but I'm ok with that.

Reply:
As for CGMManagerStatus -> CGMStatus, I probably would include any status changes related to the manager itself in this type, too, outside of the CGM hardware. But I'm fine with either name, if you still think CGMStatus is better.

Reply:
Hm...your comment about `CGMResult` seems to be an excellent case for `Result<>`:
Maybe:
```swift
// Before
public enum CGMResult {
    case noData
    case newData([NewGlucoseSample])
    case error(Error)
}
```
```swift
// After
public enum CGMError: Error {
    case noData
    case unknown(Error)
}

public typealias CGMSensorReadingResult = Result<[NewGlucoseSample], CGMError>
```

Just a suggestion...WDYT?

I don't feel as strongly about the change to `CGMManagerStatus`.  I have a slight preference for `CGMStatus`, but that's about it.

Reply:
I think that just makes for a more nested result that is harder to work with. And .noData is not really a .failure. How about renaming CGMResult to CGMReadingResult? And the method could be 
`func cgmManager(_ manager: CGMManager, hasNew readingResult: CGMReadingResult) -> Void`

Reply:
Sure, that's ok with me.